storm-core/src/jvm/backtype/storm/utils/TransferDrainer.java

```diff
@@ -26,10 +26,10 @@
 
 public class TransferDrainer {
 
-  private HashMap<String, ArrayList<ArrayList<TaskMessage>>> bundles = new HashMap();
+  private HashMap<Integer, ArrayList<ArrayList<TaskMessage>>> bundles = new HashMap();
   
-  public void add(HashMap<String, ArrayList<TaskMessage>> workerTupleSetMap) {
-    for (String key : workerTupleSetMap.keySet()) {
+  public void add(HashMap<Integer, ArrayList<TaskMessage>> taskTupleSetMap) {
```

Comment:
@d2r 
Looking it again, it should stores tuples to buffer unless batch-end? is true.
0666c41 stores tuples to ArrayList, and 861a92e just stores tuples to TransferDrainer itself.
So if we changed add() to private or remove add(), we should introduce another buffer, which role may be duplicated to TransferDrainer.
(It may not an issue when batch-end? is always true, but I don't know.)

Which one do you think is better?